packages/test-utils/package.json

```diff
@@ -54,21 +54,19 @@
     "@backstage/types": "workspace:^",
     "@material-ui/core": "^4.12.2",
     "@material-ui/icons": "^4.9.1",
-    "@testing-library/dom": "^8.0.0",
-    "@testing-library/jest-dom": "^5.10.1",
-    "@testing-library/react": "^12.1.3",
-    "@testing-library/user-event": "^14.0.0",
     "@types/react": "^16.13.1 || ^17.0.0",
     "cross-fetch": "^3.1.5",
     "zen-observable": "^0.10.0"
   },
   "peerDependencies": {
-    "react": "^16.13.1 || ^17.0.0",
-    "react-dom": "^16.13.1 || ^17.0.0",
+    "@testing-library/react": "^12.1.3 || ^13.0.0 || ^14.0.0",
+    "react": "^16.13.1 || ^17.0.0 || ^18.0.0",
+    "react-dom": "^16.13.1 || ^17.0.0 || ^18.0.0",
```

Comment:
How do you feel about automatically enabling the [legacyRoot](https://testing-library.com/docs/react-testing-library/api#legacyroot) option if we detect that we're on `react-dom` 18, by checking `require('react-dom/package.json').version`? Without that I feel it's a bit of a lie to include v18 in the version range.

Reply:
I'm following the footsteps of https://github.com/backstage/demo/pull/139 on our own conversion to React18, i.e., we're not using `legacyRoot`, but using the new React18 render setup.

That's kinda my main reason why I want to run our tests against the non-legacy rendering, too.

Reply:
Is the concern that there could a problem around automatic batching in concurrent mode?

I'd not set `legacyRoot` by default, but it's probably a good idea to expose it on `renderInTestApp` ? `renderWithEffects` inherits it from RTL, we'd just need to pipe that through.

@Rugvip, what do you think?

Reply:
Sorry I got the initial comment backwards, it would be `legacyRoot` if we detect `react-dom` v17. So that even if you have RTL v14 installed with `react-dom` v17, thinks should still work. Then again perhaps that's a bit of a moot thing because you'd need to match those two up for the rest of the project anyway.

Alright, I'd say we should be forwarding `legacyRoot` to our two render functions either way. Let's add that and then :shipit:

Reply:
RTL 13/14 doesn't support React <18, so we'll not mix the versions up.

The option is useful though if you're using 18 w/out concurrent rendering due to whatever.

I added that now. I tried to mimick the `Pick` in a few ways, but they either complained about typescripty things, or left the legacyoption w/out any useful type. We'll need to figure out a good way once we actually try to support React 18 in-tree, but for now, just hard-coding it to boolean sounded like a good-enough option.